### DISCUSSION TITLE
Problem with alignment with ul element.  Different in development and production mode.

### DISCUSSION THREAD

**Discussion by nevil2** [tailwindlabs/tailwindcss#15368] [General]
**What version of Tailwind CSS are you using?**

4.0.0-beta.6

**What build tool (or framework if it abstracts the build tool) are you using?**

vite 6.0.3, @tailwindcss/vite 4.0.0-beta.6, @sveltejs/kit 2.10.1

**What version of Node.js are you using?**

v22.11.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**Reproduction URL**

```
<ul class="list-disc">
    {#each participants as participant}
        <li>{participant}</li>
    {/each}
</ul>
```

**Describe your issue**

Problem with alignment with ul element.  Different in development and production mode.

In development mode:

![image](https://github.com/user-attachments/assets/996143fe-3813-4c99-a4ec-4b77916a724b)

In preview or production mode (after build):

![image](https://github.com/user-attachments/assets/9313d04a-4f73-4635-90dd-0dbd771ff7fd)

Development and production mode should have same formatting.  Problem is independent of browser or operating system.

**Comment by philipp-spiess** [comment]
Hey! Just tried this here with no other styles and this is working as expected for me (first screenshot is dev mode, second full build, both using Vite):

<img width="152" alt="Screenshot 2024-12-11 at 10 39 01" src="https://github.com/user-attachments/assets/854321aa-42db-430e-84b0-9c163583b7cf">
<img width="145" alt="Screenshot 2024-12-11 at 10 38 45" src="https://github.com/user-attachments/assets/52fc8369-71d1-440e-b030-cf54bc239138">

Are you able to reproduce this in a new Vite setup?

**Comment by nevil2** [comment]
I've tested with a blank SvelteKit project, with index.ccs (imported by Svelte component) as follows:

```
@import "tailwindcss";

@layer base {
  html {
    font-size: 24px;
  }

  ul {
    list-style: inside;
    list-style-type: disc;
  }
} 
```

The problem is  `list-style: inside;` which in development is interpreted as:

```
list-style-position: inside;
list-style-image: initial;
```
but is not recognised in production mode.

So the following works correctly in both modes.

```
@import "tailwindcss";

@layer base {
  html {
    font-size: 24px;
  }

  ul {
    list-style-position: inside;
    list-style-image: initial;
    list-style-type: disc;
  }
}
```
Should Tailwind support list-style: inside?  It did previously.

Can push code to GIT if you need.

**Reply by philipp-spiess** [reply]
Oh, gotcha! Thanks for the bit about `list-style: inside`. This seems to be an issue in [lightningcss](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%5Cn.foo%20%7B%5Cn%20%20list-style%3A%20inside%3B%5Cn%20%20list-style-type%3A%20disc%3B%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D). Let me file a quick bug report there so we can get this resolved upstream.

**Reply by philipp-spiess** [reply]
Filed as https://github.com/parcel-bundler/lightningcss/issues/867

**Reply by philipp-spiess** [reply]
To make this work in the meantime you can manually set `list-style-position: inside;` instead of `list-style: inside;`: https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22customMedia%22%3Atrue%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22include%22%3A0%2C%22exclude%22%3A0%2C%22source%22%3A%22%5Cn.foo%20%7B%5Cn%20%20list-style-position%3A%20inside%3B%5Cn%20%20list-style-type%3A%20disc%3B%5Cn%7D%22%2C%22visitorEnabled%22%3Afalse%2C%22visitor%22%3A%22%7B%5Cn%20%20Color(color)%20%7B%5Cn%20%20%20%20if%20(color.type%20%3D%3D%3D%20'rgb')%20%7B%5Cn%20%20%20%20%20%20color.g%20%3D%200%3B%5Cn%20%20%20%20%20%20return%20color%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D
